File: .claude/skills/agent-skill-kit-commons/packages/py/src/skills_ref/scripts/check_naming.py
from __future__ import annotations

import argparse
import json
import sys
from pathlib import Path

from ..parser import parse_frontmatter
from ..rules.naming import check_naming


def main(argv=None) -> int:
    parser = argparse.ArgumentParser(prog="check-naming")
    parser.add_argument("--fixture", required=True)
    parser.add_argument("--extra-tiers", default="")
    args = parser.parse_args(argv)
    p = Path(args.fixture).resolve()
    parsed = parse_frontmatter((p / "SKILL.md").read_text(encoding="utf-8"))
    extra = [t.strip() for t in args.extra_tiers.split(",") if t.strip()]
    issues = check_naming(parsed, p.name, {"extra_tiers": extra})
    sys.stdout.write(
        json.dumps([i.to_dict() for i in issues], sort_keys=True, separators=(",", ":"))
        + "\n"
    )
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: .claude/skills/agent-skill-kit-commons/packages/mjs/bin/check-naming.mjs
#!/usr/bin/env node
import { readFileSync } from "node:fs";
import { join, basename, resolve } from "node:path";
import { parseFrontmatter } from "../src/parser.mjs";
import { checkNaming } from "../src/rules/naming.mjs";

function parseArgs(argv) {
  const opts = { fixture: null, extraTiers: [] };
  for (let i = 0; i < argv.length; i++) {
    if (argv[i] === "--fixture") opts.fixture = argv[++i];
    else if (argv[i] === "--extra-tiers")
      opts.extraTiers = (argv[++i] ?? "")
        .split(",")
        .map((s) => s.trim())
        .filter(Boolean);
  }
  return opts;
}

const { fixture, extraTiers } = parseArgs(process.argv.slice(2));
if (!fixture) {
  process.stderr.write("usage: check-naming --fixture <path>\n");
  process.exit(64);
}
const absDir = resolve(fixture);
const skillPath = join(absDir, "SKILL.md");
const text = readFileSync(skillPath, "utf8");
const parsed = parseFrontmatter(text);
const dirName = basename(absDir);
const issues = checkNaming(parsed, dirName, { extraTiers });
function sortKeys(v) {
  if (Array.isArray(v)) return v.map(sortKeys);
  if (v && typeof v === "object") {
    const out = {};
    for (const k of Object.keys(v).sort()) out[k] = sortKeys(v[k]);
    return out;
  }
  return v;
}
process.stdout.write(JSON.stringify(sortKeys(issues)) + "\n");
